### PR TITLE
ECM-434 Fix Case API updates MultipleReferenceLinkMarkUp

### PR DESCRIPTION
**Pull request by cindychmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-434

### Change description ###

Extend Fix Case API to update MultipleReferenceLinkMarkUp in single case

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public void checkUpdateMultipleReference(CaseDetails caseDetails, String userToken) {
+        CaseData caseData = caseDetails.getCaseData();
+        if (caseData.getEcmCaseType().equals(MULTIPLE_CASE_TYPE)) {
+            String multipleCaseTypeId = UtilHelper.getBulkCaseTypeId(caseDetails.getCaseTypeId());
+            List<SubmitMultipleEvent> submitMultipleEvents = multipleCasesReadingService.retrieveMultipleCases(
+                    userToken, multipleCaseTypeId, caseData.getMultipleReference());
+            if (!submitMultipleEvents.isEmpty()) {
+                String multipleCaseId = String.valueOf(submitMultipleEvents.get(0).getCaseId());
+                caseData.setMultipleReferenceLinkMarkUp(getLinkMarkUp(multipleCaseId, caseData.getMultipleReference()));
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:34]: I think this means you need to change ecm-common. The ES query is using the field name "reference" but you are not passing the CCD reference. Instead you are passing multipleReference (this is a reference generated by ECM). Therefore, you need to change the ES query to filter on that field

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class FixCaseApiService {
+
+    private final MultipleCasesReadingService multipleCasesReadingService;
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    public void checkUpdateMultipleReference(CaseDetails caseDetails, String userToken) {
+        CaseData caseData = caseDetails.getCaseData();
+        if (caseData.getEcmCaseType() != null && caseData.getEcmCaseType().equals(MULTIPLE_CASE_TYPE)) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java:28]: You can improve this a little bit. The null check isn't required if you do this:
`if (MULTIPLE_CASE_TYPE.equals(caseData.getEcmCaseType(() {`

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java [hunk 3]
```diff
@@ -1069,6 +1073,15 @@ public void judgmentSubmittedError400() throws Exception {
                 .andExpect(status().isBadRequest());
     }
 
+    @Test
+    public void dynamicFixCaseAPIError400() throws Exception {
+        mvc.perform(post(DYNAMIC_FIX_CASE_API_URL)
+                .content("error")
+                .header("Authorization", AUTH_TOKEN)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java:1082]: You can improve the 400/500 error unit tests by also checking a service wasn't called. See mockito's verify/never
e.g. something like this
`verify(fixCaseApiService, never()).checkUpdateMultipleReference(any(CaseDetails.class), anyString());`

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java (hunk 2 above)
Comment (cindychmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java]: Updated

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java (hunk 3 above)
Comment (cindychmcts) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerControllerTest.java:1086]: Updated

**Review by jthmcts — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**